src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy

```diff
@@ -240,14 +253,31 @@ public enum MysteriousTreasures implements LogicCardInfo {
           pokePower "Power Cancel", {
             text "Once during your opponent’s turn, when your opponent’s Pokémon uses any Poké-Power, you may discard 2 cards from your hand and prevent all effects of that Poké-Power. (This counts as that Pokémon using its Poké-Power.) This power can’t be used if Alakazam is affected by a Special Condition."
             actionA {
+              //TODO: Yeah this is gonna be a fun one to do.
+              //Requires a way to capture "powerUsed", stop the Poké-Power at that point but make it count as used.
```

Comment:
Based off the usual "Is ability blocked" code here:
```
eff = getter IS_ABILITY_BLOCKED, { Holder h->
            if (self.active && (h.effect.ability instanceof PokeBody || h.effect.ability instanceof PokePower)) {
              h.object=true
            }
          }
```
I wonder if you can just set `h.effect.ability.used = true` where you set the block to true as well.

Reply:
I think it might require it's own handler (wanna talk about it with axpendix later if possible. Main issue is making sure the opponent _can_ use the power, and then as soon as it's marked as powerUsed() trigger the popup for activating this blocker PokéPower. 

It *could* just pop up once the power button is clicked, like I understand that one works. But I'd like 100% confirmation before proceeding with it. If it blocks as soon as the button is pressed, the opponent could just call a Power they know they can't use, get that one blocked and use the rest normally (since it's a single block per ability/card).